src/crc32.c

```diff
@@ -132,7 +132,14 @@ bool generic_crc32(target_s *const target, uint32_t *const result, const uint32_
 		for (size_t i = 0; i < read_len; i++)
 			crc = crc32_calc(crc, bytes[i]);
 	}
-	DEBUG_WARN("%" PRIu32 " ms\n", platform_time_ms() - start_time);
+#if ENABLE_DEBUG == 1
+	const uint32_t end_time = platform_time_ms();
+	if (len > 512U) {
+		const uint32_t speed = len * 1000U / (end_time - start_time) / 1024U;
+		DEBUG_WARN("len = %" PRIu32 ": %" PRIu32 " KiB/s,\t", (uint32_t)len, speed);
+	}
+	DEBUG_WARN("%" PRIu32 " ms\n", end_time - start_time);
```

Comment:
Let's prefix this and the one below with some context? Otherwise we still have the same problem as before where "20ms" printed on the console without any further information is just meaningless fluff due to the lack of context.

Perhaps:
```suggestion
	DEBUG_WARN("crc32: %08" PRIx32 "+%" PRIu32 " -> %" PRIu32 " ms", base, (uint32_t)len, end_time - start_time);
	if (len > 512) {
		const uint32_t speed = len * 1000U / (end_time - start_time) / 1024U;
		DEBUG_WARN(" %" PRIu32 " KiB/s", speed);
	}
	DEBUG_WARN("\n");
```

Reply:
These random "1ms" lines on BMDA stdout log always looked funny to me. Why not `__func__` for generic_crc32: ? This does not get into release BMF builds (yet), and so 8 more bytes is not a space concern.
Do you approve of the 512 section length lower bound? Also I successfully avoided 64-bit integer division here, and I don't think it will overflow practically (biggest targets have 16 MiB of flash on e.g. RP2040), but precision is limited in a way.

Reply:
Those random lines is exactly what we're aiming to avoid 👍🏼 
We could use `__func__`, but well, writing it this way means it's identical between implementations which we think is a good thing with the context.

We weren't able to fully validate the division, but it seems reasonable and the bound while arbitrary seems fine. As it's intended as a lower bound, then perhaps `>=` instead of just `>`?

Reply:
The way BMD is organized means both these functions have to have identical names and signatures to link (inverse of ODR). I wonder if using attribute alias would allow renaming the second one to `stm32_hwcrc32()` or something alike.
Noted the `len >= 512U` suggestion. Users could look at gdb console output and be able to match hex section bounds with BMD ms-timing logs for smaller regions.

Reply:
Attribute alias certainly would allow the function name to reflect what it is properly - that might be a fair idea to use here! Perhaps `bmd_crc32()` for the base function name, `generic_crc32()` for the non-STM32-specific one and `stm32_crc32()` for the specific one, all aliased to `bmd_crc32()`?

Reply:
Yes, I tried it and that works, that is it gets into the binary and `strings -n8 src/blackmagic.bin` (similar for stripped hosted) only lists the actual name from supposedly string table resources, but never `bmd_crc32`. And there's only a single callsite in gdb_main, so this PR can still be kept small scope.

Reply:
oh yeah, given those other errors were using the function name, that seems absolutely fine then too to use __func__ and save back a touch of space overall